home/views/api/leaderboard.py

```diff
@@ -0,0 +1,365 @@
+
+#     # Check if url has any contest parameters
+#     contest_id = self.request.GET.get("contest_id")
+
+#     # Check if url has any contest parameters
+#     include_testers = self.request.GET.get("include_testers") is not None
+
+#     # Default href for download button
+#     context["download_url"] = "/data/users_agg.csv"
+
+#     contest = (
+#         Contest.objects.get(contest_id=contest_id) if contest_id else None
+#     )
+#     daily_walks, intentional_walks, _, _ = get_contest_walks(contest)
+
+#     # Prepare loading of data into context
+#     user_stats_container = {}
+#     context["user_stats_list"] = []
+
+#     # If a contest is specified, include all new signups, regardless of
+#     # whether they walked during the contest or not.
+#     if contest is not None:
+#         # Context variables
+#         context["current_contest"] = contest
+#         context["download_url"] += f"?contest_id={contest_id}"
+
+#         # # Find everyone who signed up during the constest
+#         # for acct in get_new_signups(contest, include_testers):
+#         #     if acct["email"] not in daily_walks:
+#         #         user_stats_container[acct["email"]] = dict(
+#         #             new_signup=True, account=acct
+#         #         )
+#         #         new_signups_by_zip[acct["zip"]] += 1
+
+#     # Add all accounts found in filtered daily walk data
+#     for email, dw_row in daily_walks.items():
+#         acct = Account.objects.values(*ACCOUNT_FIELDS).get(email=email)
+
+#         # Skip testers unless include_testers
+#         if not include_testers and acct.get("is_tester"):
+#             continue
+
+#         # Don't include those who signed up after the contest ended
+#         if contest and acct["created"] > localize(contest.end):
+#             continue
+
+#         user_stats = user_stats_container.get(
+#             email, dict(new_signup=False)
+#         )
+#         user_stats["account"] = acct
+#         user_stats["dw_steps"] = dw_row["dw_steps"]
+
+#         # Also add recorded walk data
+#         iw_row = intentional_walks.get(email)
+#         if iw_row:
+#             user_stats["rw_steps"] = iw_row["rw_steps"]
+#             user_stats["rw_distance"] = iw_row["rw_distance"]
+#             user_stats["rw_time"] = (
+#                 iw_row["rw_total_walk_time"].total_seconds()
+#                 - iw_row["rw_pause_time"]
+#             ) / 60  # minutes
+#         user_stats["num_rws"] = iw_row["rw_count"] if iw_row else 0
+
+#         # Put user_stats (row) back into container
+#         user_stats_container[email] = user_stats
+
+#     context["user_stats_list"] = user_stats_container.values()
+#     context["contests"] = Contest.objects.all()
+
+#     return context
+
+
+#
+@method_decorator(csrf_exempt, name="dispatch")
+class LeaderboardListView(View):
+    """View to retrieve leaderboard"""
+
+    # model = DailyWalk
+    http_method_names = ["post"]
```

Comment:
POST is typically used when a client is sending data to a server for creating/updating records.

Since this is just a fetch/read of data, GET would probably be a more appropriate HTTP protocol action verb/method. Then, instead of expecting a JSON body, extract any necessary params from query string parameters (i.e. `?account_id=...` in the url).